crates/core/src/lib.rs

```diff
@@ -64,7 +64,12 @@ pub unsafe extern "C" fn compile_src(js_src_ptr: *const u8, js_src_len: usize) -
 pub unsafe extern "C" fn eval_bytecode(bytecode_ptr: *const u8, bytecode_len: usize) {
     let runtime = RUNTIME.get().unwrap();
     let bytecode = slice::from_raw_parts(bytecode_ptr, bytecode_len);
-    execution::run_bytecode(runtime, bytecode).unwrap();
+    match execution::run_bytecode(runtime, bytecode) {
+        Ok(_) => {}
+        Err(e) => {
+            eprintln!("Error while running JS: {}", e);
+        }
```

Comment:
We do want the error case to trigger a trap since we want code that has failed to run to signal that it has failed to run to whatever is invoking it. However, we don't want a panic because that adds a lot of noise to the logs that we do not want. Using `std::process::abort()` is one option. Another is invoking `std::process::exit` with the canonical `FAILURE` `std::process::ExitCode` value.

Also is there a reason not to move this logic into the `execution` module given the code here and in `main.rs` will be identical?

Final nit, you can simplify the `match` statement to an `or_else` statement given the `Ok` branch is not performing any operations.